crates/validation_common/src/lib.rs

```diff
@@ -0,0 +1,131 @@
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize)]
+#[serde(deny_unknown_fields)]
+struct UnvalidatedMeasurements {
+    #[serde(default, rename = "enarx_signer")]
+    signer: Vec<Digest>,
+
+    #[serde(default, rename = "enarx_hash")]
+    hash: Vec<Digest>,
+}
+
+impl TryFrom<UnvalidatedMeasurements> for Measurements {
+    type Error = &'static str;
+
+    fn try_from(
+        UnvalidatedMeasurements { signer, hash }: UnvalidatedMeasurements,
+    ) -> Result<Self, Self::Error> {
+        if signer.is_empty() && hash.is_empty() {
+            Err("one of `enarx_signer` or `enarx_hash` must be specified")
+        } else {
+            Ok(Self { signer, hash })
+        }
+    }
+}
+
+#[derive(Clone, Debug, Default, Serialize, Eq, PartialEq, Deserialize)]
+#[serde(deny_unknown_fields, try_from = "UnvalidatedMeasurements")]
+pub struct Measurements {
+    /// Allowed signing key digests
+    #[serde(rename = "enarx_hash")]
+    pub signer: Vec<Digest>,
+
+    /// Allowed Enarx binary digests.
+    /// This is the hash of the Enclave environment after the Enarx binary is loaded
+    /// but before any workload is loaded, so this is a hash of the Enarx binary
+    /// in memory.
+    #[serde(rename = "enarx_hash")]
+    pub hash: Vec<Digest>,
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::*;
+
+    #[test]
+    fn deserialize_hash_to_vec() {
+        #[derive(Clone, Debug, Default, Deserialize)]
+        struct TestConfig {
+            some_hash: Digest,
+        }
+
+        let config: TestConfig = toml::from_str(
+            r#"
+        some_hash = "0102030405060708090a0b0c0d"
+        "#,
+        )
+        .expect("Couldn't deserialize");
+        assert_eq!(
+            config.some_hash.0,
+            [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13]
+        );
+    }
+
+    #[test]
+    fn test_signer() {
+        let measurements: Measurements = toml::from_str(
+            r#"
+        enarx_signer = ["1234567890", "00112233445566778899"]
+        "#,
+        )
+        .expect("Couldn't deserialize");
+
+        assert_eq!(measurements.signer.len(), 2);
+        assert_eq!(
+            measurements.signer.get(0).unwrap().as_ref(),
+            &hex::decode("1234567890").unwrap()
+        );
+        assert_eq!(
+            measurements.signer.get(1).unwrap().as_ref(),
+            &hex::decode("00112233445566778899").unwrap()
+        );
+    }
+
+    #[test]
+    fn test_hash() {
+        let measurements: Measurements = toml::from_str(
+            r#"
+        enarx_hash = ["1234567890", "00112233445566778899"]
+        "#,
+        )
+        .expect("Couldn't deserialize");
+
+        assert_eq!(measurements.hash.len(), 2);
+        assert_eq!(
+            measurements.hash.get(0).unwrap().as_ref(),
+            &hex::decode("1234567890").unwrap()
+        );
+        assert_eq!(
+            measurements.hash.get(1).unwrap().as_ref(),
+            &hex::decode("00112233445566778899").unwrap()
+        );
+    }
```

Comment:
Please use the tests I provided in the diff, this test suite is incomplete and hard to maintain.